deployment/vm/manager.go

```diff
@@ -94,9 +94,32 @@ func (m *manager) Create(stemcell bistemcell.CloudStemcell, deploymentManifest b
 		return nil, bosherr.WrapErrorf(err, "Getting resource pool for job '%s'", jobName)
 	}
 
-	agentID, err := m.uuidGenerator.Generate()
+	agentID, found, err := m.vmRepo.FindCurrentAgentId()
 	if err != nil {
-		return nil, bosherr.WrapError(err, "Generating agent ID")
+		return nil, bosherr.WrapError(err, "Finding currently agent id of deployed vm")
+	}
+
+	if !found {
+		agentID, err = m.uuidGenerator.Generate()
+		if err != nil {
+			return nil, bosherr.WrapError(err, "Generating agent ID")
+		}
+	}
+
+	currentIP, found, err := m.vmRepo.FindCurrentIP()
+	if err != nil {
+		return nil, bosherr.WrapError(err, "Finding currently IP address of deployed vm")
+	}
+
+	ifaceMap := map[string]biproperty.Map{}
+
+	if found {
```

Comment:
Can we move the case for `found` and not `found` into private methods so that this one is easier to read?